src/data_info/services/stat.clj

```diff
@@ -46,76 +65,107 @@
-      :dir-count  (icat/number-of-folders-in-folder user (cfg/irods-zone) path))
+      :file-count (if (needs-key? included-keys :file-count) (icat/number-of-files-in-folder user (cfg/irods-zone) path))
+      :dir-count  (if (needs-key? included-keys :dir-count)  (icat/number-of-folders-in-folder user (cfg/irods-zone) path)))
     stat-map))
 
 
 (defn- merge-shares
-  [stat-map cm user path]
-  (if (owns? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-key? included-keys :share-count) (owns? stat-map))
     (assoc stat-map :share-count (count-shares cm user path))
     stat-map))
 
 
 (defn- merge-label
-  [stat-map user path]
-  (assoc stat-map
-         :label (paths/path->label user path)))
+  [stat-map user path included-keys]
+  (if (needs-key? included-keys :label)
+    (assoc stat-map
+           :label (paths/path->label user path))
+    stat-map))
 
 
 (defn- merge-type-info
-  [stat-map cm user path]
-  (if-not (is-dir? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-any-key? included-keys :infoType :content-type) (not (is-dir? stat-map)))
     (assoc stat-map
-      :infoType     (get-types cm user path)
-      :content-type (irods/detect-media-type cm path))
+      :infoType     (if (needs-key? included-keys :infoType) (get-types cm user path))
+      :content-type (if (needs-key? included-keys :content-type) (irods/detect-media-type cm path)))
     stat-map))
 
-
 (defn ^IPersistentMap decorate-stat
-  [^IPersistentMap cm ^String user ^IPersistentMap stat]
+  [^IPersistentMap cm ^String user ^IPersistentMap stat included-keys]
   (let [path (:path stat)]
     (-> stat
-      (assoc :id         (-> (meta/get-attribute cm path uuid/uuid-attr) first :value)
-             :permission (perm/permission-for cm user path))
-      (merge-label user path)
-      (merge-type-info cm user path)
-      (merge-shares cm user path)
-      (merge-counts cm user path))))
-
+      (assoc :id         (if (needs-key? included-keys :id) (-> (meta/get-attribute cm path uuid/uuid-attr) first :value))
+             :permission (if (needs-key? included-keys :permission) (perm/permission-for cm user path)))
+      (merge-label user path included-keys)
+      (merge-type-info cm user path included-keys)
+      (merge-shares cm user path included-keys)
+      (merge-counts cm user path included-keys)
+      (select-keys included-keys))))
+
+(defn- get-filter-set
+  [filter-vec-or-string default]
+  (if (nil? filter-vec-or-string)
+    (set default)
+    (if (string? filter-vec-or-string)
+      (set (map keyword (string/split filter-vec-or-string #",")))
+      (set (map keyword filter-vec-or-string)))))
+
+(defn- process-filters
+  "Process an include and an exclude string into just a list of keys to include"
+  [include exclude]
+  (let [all-keys #{:id :path :type :label :date-created :date-modified :permission :share-count :file-count :dir-count :file-size :content-type :infoType :md5}
```

Comment:
Sounds like a plan. Possibly I'll also see if I can find a way to validate more thoroughly, but it seems like swagger and compojure-api might disagree about how to pass arrays to endpoints, so it might or might not work, heh.